api/cas/mixins.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+import jwe
+import jwt
+
+from rest_framework.exceptions import ParseError
+
+from api.base import settings
+from api.cas import messages
+
+from framework import sentry
+
+
+class APICASMixin(object):
+    """ Mixin Class for API CAS Views.
+    """
+
+    def load_request_body_data(self, request):
```

Comment:
The Mixin Class for API CAS views and authentication classes. The primary goal is to decode and decrypt CAS request body and return the body data. The `load_request_body_data()` is run in the authentication class for views that has one and in the view itself if not.

We used `middleware` in early design, but exceptions cannot be handled gracefully there. Every exception returns 500 instead of dedicated 40x CAS expects.